src/js/libgeo/GeoBasics.js

```diff
@@ -174,9 +195,13 @@ function addElementNoProof(el) {
                 return null;
             }
             if (op.signature !== "**") {
+                var opSigs;
+                if (isSet) {
+                    opSigs = el.args.map(getKind);
```

Comment:
This feels kind of weird. If I understand you correctly, you generate a list of actual kinds to compare against the actual kinds. A check which will never fail. So I think it should be skipped in the first place. The code here is also inside the `for i` loop, so you do this again for each argument, which comes at a slight performance cost and at some potential for confusion as well. Can't you just add `|| !isSet` to the `**` check above? Or, in case you decide on `*` instead of `s` as the suffix, make that `op.signature.charAt(1) !== "*"` or `/[^*]$/.test(op.signature)` or something like that?

Another alternative would be moving the semantics from the `el.args.forEach` above to this `for (i…` loop here. In the code above you'd set `isSet` but not do any type checking. Then here you'd set `opSig` to `op.signatures[i]` if you are in the `!isSet` case, but to `op.signature.charAt(1)` in the `isSet` case. Then you compare `opSig` against `argKind`. That would concentrate all the looping over arguments to a single location, which I think would make this easier to maintain.